MIGRATION.md

```diff
@@ -0,0 +1,155 @@
+IP/port or a host name/port combination. When a request comes in, IIS matches
+the port, determines the host name from the request, and searches the
+centralized certificate store for a certificate file with a matching name. It
+uses that certificate.
+
+With a Server Name Indicator (SNI), the host name is exchanged as part of the
+SSL handshake. SNI is enabled in the Add Site Binding dialog box when you add a
+binding with a type of HTTPS. This is especially useful for SSL connections
+that host multiple servers on a single network address. For more information,
+see IIS 8.0 Server Name Indication (SNI): SSL Scalability.
+
+The sslFlags attribute is only set when the protocol is https.
+
+The default value is 0.
+#### Parameter mapping:
+puppet-iis  | puppetlabs-iis
+------------|--------------
+ensure      | identical
+name        | identical
+path        | physicalpath
+app_pool    | applicationpool
+host_header | bindings hostname*
+protocol    | bindings protocol*
+ip          | bindings bindinginformation*
+port        | bindings bindinginformation*
+ssl         | bindings sslflags*
+
+\* The `bindings` parameter on the new `iis_site` take an array of one or more bindings as a hash and have the following keys per hash:
+- protocol
+- bindinginformation
+- sslflags
+
+The bindinginformation is in the form `<ip>:<port>`
+
+### `iis::manage_binding`
+All of these properties are managed directly by the `iis_site::bindings` property and are not declared separately, as the `iis_site` and `iis::manage_binding` resources are. The following describe the `iis_site::bindings` hash keys for the mapping:
+
+#### Parameter mapping:
+puppet-iis             | puppetlabs-iis
+-----------------------|--------------
+site_name              | not needed
+ensure                 | not needed
+require_site           | not needed
+protocol               | identical
+host_header            | hostname
+ip_address             | first half of bindinginformation
+port                   | last half of bindinginformation
+certificate_thumbprint | certificatehash
+store                  | certificatestorename
+ssl_flag               | sslflags
+
+### `iis_application`
+This type is still called `iis_application`
+
+#### Parameter mapping:
+puppet-iis | puppetlabs-iis
+-----------|--------------
+ensure     | identical
+name       | identical
+path       | physicalpath
+site       | sitename *
+app_pool   | applicationpool
+
+\* `site` is a reserved word in puppet 4 and will cause failures, so `sitename` is the new name.
+
+## Unmapped functionality
+### Class iis
+Declares six groups of features via `iis::features::*` to allow them to be easily installed. This functionality is available via `iis_feature` but must be explicitly declared.
+
+The list of feature groups are:
+- application deployment
+- common http
+- health and diagnostics
+- management tools
+- performance
+- security
+
+
+### `iis_virtualdirectory`
+MISSING! This will be added with the proposed changes:
```

Comment:
Is it the parameter mapping that are the proposed changes? When and how do are those changes happening? I'm confused by this one.